ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/MetadataCache.java

```diff
@@ -17,32 +17,27 @@
  */
 package org.apache.hadoop.hive.ql.txn.compactor;
 
-import com.google.common.cache.Cache;
-import com.google.common.cache.CacheBuilder;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
 import org.apache.thrift.TBase;
 
-import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
 
 public class MetadataCache {
 
   private Cache<String, TBase> metaCache;
 
   public MetadataCache(boolean tableCacheOn) {
     if (tableCacheOn) {
-      metaCache = CacheBuilder.newBuilder().softValues().build();
+      metaCache = Caffeine.newBuilder().softValues().build();
     }
   }
 
-  public <T extends TBase<T,?>> T computeIfAbsent(String key, Callable<T> callable) throws Exception {
+  public <T extends TBase<T,?>> T computeIfAbsent(String key, Function<? super String, ? extends TBase> function) throws Exception {
     if (metaCache != null) {
-      try {
-        return (T) metaCache.get(key, callable);
-      } catch (ExecutionException e) {
-        throw (Exception) e.getCause();
-      }
```

Comment:
Why did you replace the exception handling with a Functional interface? Is there any benefit from it, because now the client code must wrap each call.

Reply:
This is required as an input parameter for caching in Caffeine is - 
`metaCache.get(key, an instance of Function).`

Since the API itself is using functional interface, I modified all callables to functional instances. Yes there is a problem of exception handling when using functional interface which is why I wrote the `ThrowingFunction` interface.